cli/azd/pkg/azdo/pipeline.go

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azdo
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/microsoft/azure-devops-go-api/azuredevops"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/build"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/taskagent"
+)
+
+// Creates a variable to be associated with a Pipeline
+func createBuildDefinitionVariable(value string, isSecret bool, allowOverride bool) build.BuildDefinitionVariable {
+	return build.BuildDefinitionVariable{
+		AllowOverride: &allowOverride,
+		IsSecret:      &isSecret,
+		Value:         &value,
+	}
+}
+
+// returns the default agent queue. This is used to associate a Pipeline with a default agent pool queue
+func getAgentQueue(ctx context.Context, projectId string, connection *azuredevops.Connection) (*taskagent.TaskAgentQueue, error) {
+	client, err := taskagent.NewClient(ctx, connection)
+	if err != nil {
+		return nil, err
+	}
+	getAgentQueuesArgs := taskagent.GetAgentQueuesArgs{
+		Project: &projectId,
+	}
+	queues, err := client.GetAgentQueues(ctx, getAgentQueuesArgs)
+	if err != nil {
+		return nil, err
+	}
+	for _, queue := range *queues {
+		if *queue.Name == "Default" {
+			return &queue, nil
+		}
+	}
+	return nil, fmt.Errorf("could not find a default agent queue in project %s", projectId)
+}
+
+// create a new Azure DevOps pipeline
+func CreatePipeline(
+	ctx context.Context,
+	projectId string,
+	name string,
+	repoName string,
+	connection *azuredevops.Connection,
+	credentials AzureServicePrincipalCredentials,
+	env environment.Environment) (*build.BuildDefinition, error) {
+
+	client, err := build.NewClient(ctx, connection)
+	if err != nil {
+		return nil, err
+	}
+
+	repoType := "tfsgit"
+	buildDefinitionType := build.DefinitionType("build")
+	definitionQueueStatus := build.DefinitionQueueStatus("enabled")
+	defaultBranch := fmt.Sprintf("refs/heads/%s", DefaultBranch)
+	buildRepository := &build.BuildRepository{
+		Type:          &repoType,
+		Name:          &repoName,
+		DefaultBranch: &defaultBranch,
+	}
+
+	process := make(map[string]interface{})
+	process["type"] = 2
+	process["yamlFilename"] = AzurePipelineYamlPath
+
+	variables := make(map[string]build.BuildDefinitionVariable)
+	variables["AZURE_SUBSCRIPTION_ID"] = createBuildDefinitionVariable(credentials.SubscriptionId, false, false)
+	variables["ARM_TENANT_ID"] = createBuildDefinitionVariable(credentials.TenantId, false, false)
+	variables["ARM_CLIENT_ID"] = createBuildDefinitionVariable(credentials.ClientId, true, false)
```

Comment:
Is ARM_ a new thing?  Why not AZURE_?

Reply:
These names are specifically needed for the Terraform azurerm provider.